packages/callhierarchy/src/browser/callhierarchy-service-impl.ts

```diff
@@ -239,49 +159,78 @@ export abstract class AbstractDefaultCallHierarchyService implements CallHierarc
      * As we only check regions that contain the definition, that is the one with the
      * latest start position.
      */
-    protected async getEnclosingRootSymbol(definition: Location, context: CallHierarchyContext): Promise<SymbolInformation | undefined> {
-        const symbols = (await context.getAllSymbols(definition.uri)).filter(s => this.isCallable(s));
-        let bestMatch: SymbolInformation | undefined = undefined;
-        let bestRange: Range | undefined = undefined;
-        for (const candidate of symbols) {
-            const candidateRange = candidate.location.range;
-            if (utils.containsRange(candidateRange, definition.range)) {
-                if (!bestMatch || utils.startsAfter(candidateRange, bestRange!)) {
-                    bestMatch = candidate;
-                    bestRange = candidateRange;
+    protected async getEnclosingRootSymbol(definition: Location, context: CallHierarchyContext): Promise<ExtendedDocumentSymbol | SymbolInformation | undefined> {
+        const allSymbols = await context.getAllSymbols(definition.uri);
+        if (allSymbols.length === 0) {
+            return undefined;
+        }
+        if (DocumentSymbol.is(allSymbols[0])) {
+            const symbols = (allSymbols as DocumentSymbol[]);
+            const containsDefinition = (symbol: DocumentSymbol) => utils.containsRange(symbol.range, definition.range);
```

Comment:
'would have loved to use `DocumentSymbol.nameRange` here, it would render it to match test.